x-pack/plugins/apm/public/components/fleet_integration/apm_policy_form/typings.ts

```diff
@@ -19,7 +20,33 @@ export {
 
 export type PackagePolicyVars = Record<string, PackagePolicyConfigRecordEntry>;
 
-export type OnFormChangeFn = (
-  newVars: PackagePolicyVars,
-  isValid: boolean
-) => void;
+export type SettingValidation = t.Type<any, string, unknown>;
+
+interface AdvancedSettings {
+  type: 'advanced_settings';
+  settings: SettingDefinition[];
+}
+
+export interface Setting {
+  type:
+    | 'text'
+    | 'combo'
+    | 'area'
+    | 'boolean'
+    | 'integer'
+    | 'bytes'
+    | 'duration';
+  key: string;
+  rowTitle?: string;
+  rowDescription?: string;
+  label?: string;
+  helpText?: string;
+  placeholder?: string;
+  labelAppend?: string;
+  settings?: SettingDefinition[];
+  validation?: SettingValidation;
+  required?: boolean;
+  readOnly?: boolean;
+}
+
+export type SettingDefinition = Setting | AdvancedSettings;
```

Comment:
There's some naming here that I feel could be made more clear. What's a `Setting` and what's a `SettingDefinition`?

Contrasts can be nice. So if we have to types of settings, and one of them is advanced, the other must be basic. Then, I assume the union of those is a `Setting`:

```suggestion
export type Setting = BasicSetting | AdvancedSettings;
```

Reply:
Or perhaps we are actually talking about rows here?

```suggestion
export type SettingsRow = BasicSettingsRow | AdvancedSettingsRow;
```